Verdict for scoring problems (#161)
Kattis/problem-package-format: spec/2023-07-draft.md

The score of the group then results in the minimum or the sum of all the sub-scores, respectively.
If using `min` aggregation, then the verdict must not be accepted if any sub-verdict is rejected.
If using `sum` aggregation, then the verdict must be accepted if any sub-verdict is accepted.

The score of the submission is its score on the topmost test group `data`.

Ultimately, the score cannot be non-zero if the topmost verdict is not accepted. 
Generally, an accepted verdict will be accompanied by a strictly positive score. 
However, note that there can be a distinction between a non-accepted verdict and an accepted verdict with score 0.
An accepted verdict with a score of 0 could mean that the output is well formed but not good enough to score points.

For example, consider an optimization problem such as the Travelling Salesperson Problem.
The goal is to provide the shortest possible route, and score is determined by how short of a route the submission outputs.
If a submission instead outputs the longest possible route, it may get a verdict of accepted, because the output is well formed.
However, the route is so long that by most metrics, the submission's score would be 0.

> In most cases, the output validator could simply check whether the score is 0 and reject instead of accepting for the test case,

Reviewer: I'm not sure if I agree with this statement exactly. Maybe something like this?:
```suggestion
In most cases, an output validator should probably reject a test case rather than accepting and returning a score of 0,
```
It basically means the same thing...
